### DISCUSSION TITLE
where and for what is CADDY_SERVER_EXTRA_DIRECTIVES used for ?

### DISCUSSION THREAD

**Discussion by michabbb** [php/frankenphp#526] [Q&A]
hi there, 
I would be happy if someone could show me a useful example of how this variable can be used.
(because I cannot find any example out there)

thanks !

**Comment by dunglas** [comment]
This can be used to inject extra Caddy directives in the server bloc. Like `header` or `skip_log`: https://caddyserver.com/docs/caddyfile/directives

**Reply by michabbb** [reply]
thanks a lot for your feedback!
i am asking because I have to add something into the original Caddyfile of your frankenphp docker image.
below `{$CADDY_SERVER_EXTRA_DIRECTIVES}` or before `php_server` I have to add this:

```
handle_path /docs/* {
           root * public/docs
           try_files {path} {path}/index.php /index.php?{query}
           file_server
        }
```

because I don´t see a way to add this snippet from outside into the caddyfile, I copied the caddyfile from the docker image, changed it, and mounted it back in again:

```yaml
volumes:
  - ${DOCKER_HOME}/Caddyfile:/etc/caddy/Caddyfile:ro
```

that´s, of course, working, but I hoped there is a way to include a file as a snippet ;)
I guess this is more a docker question than a frankenphp question, but if you have any suggestions, I would be happy to hear them :)

maybe something like this:

```bash
export CADDY_SERVER_EXTRA_DIRECTIVES=$(cat snippet.txt)
docker-compose up
```

and... because of readability, I want to avoid setting a string with tons of "\n\n\n" into an ENV variable 😏 

🤔
